### PR TITLE
Enable Horizontal Autoscaling configuration in marketplace offer

### PR DESCRIPTION
**Pull request by galiacheng**
Changes
======================================

- Add **Configure Horizontal Autoscaling** blade to offer UI.
- Support HPA based on user inputs.
- Enable WebLogic Monitoring Exporter, Azure Managed Prometheus, and KEDA based on user inputs.
- Output KEDA scaler sample.

Note: have to resolve conflict after #298 is merged.

UI definition
======================================

image

image

Design diagram
======================================

image

Document
======================================
https://aka.ms/wlsaks-autoscaling

Test
======================================
https://github.com/galiacheng/weblogic-azure/actions/runs/8181432300

### CODE REVIEW

File: weblogic-azure-aks/src/main/arm/scripts/setupWLSDomain.sh [hunk 1]
```diff
@@ -383,13 +383,13 @@ function query_acr_credentials() {
     # to mitigate error in https://learn.microsoft.com/en-us/answers/questions/1188413/the-resource-with-name-name-and-type-microsoft-con
```
Comment (galiacheng) [inline weblogic-azure-aks/src/main/arm/scripts/setupWLSDomain.sh:383]: Fix the ACR not found issue by registering resource type and specifying resource group name.

**Comment by galiacheng** [comment]
Changes on 0328:
- Upload diagram.
- Resolve conflicts.
- Enhance UI.
- Resolve live issue: module 'azure.core.pipeline.policies' has no attribute 'SensitiveHeaderCleanupPolicy'

**Comment by galiacheng** [comment]
Hello @edburns I marked the PR as draft as I am working around a live site issue, see dd-2012351. The issue effects the scaling module, we don't need a hotfix currently.

File: weblogic-azure-aks/src/main/arm/scripts/createAppGatewayIngress.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2021, Oracle Corporation and/or its affiliates.
+# Copyright (c) 2021, 2024, Oracle Corporation and/or its affiliates.
```
Comment (edburns) [inline weblogic-azure-aks/src/main/arm/scripts/createAppGatewayIngress.sh:1]: Great catch.

File: weblogic-azure-aks/src/main/arm/scripts/inline-scripts/enablePrometheusMetrics.sh [hunk 3]
```diff
@@ -0,0 +1,411 @@
+# Copyright (c) 2024, Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+function connect_aks(){
+    az aks get-credentials \
+        -n $AKS_CLUSTER_NAME \
+        -g $AKS_CLUSTER_RG_NAME \
+        --overwrite-existing \
+        --only-show-errors
+}
```
Comment (edburns) [inline weblogic-azure-aks/src/main/arm/scripts/inline-scripts/enablePrometheusMetrics.sh:10]: The following scripts seem to all have a need for this same functionality. If this is common functionality, why then is it not in `utility.sh`? There needs to be a good reason for this DRY violation.

```
az aks get-credentials \
./weblogic-azure-aks/src/main/arm/scripts/inline-scripts/queryDomainConfigurations.sh
    az aks get-credentials \
./weblogic-azure-aks/src/main/arm/scripts/inline-scripts/validateAgic.sh
    az aks get-credentials \
./weblogic-azure-aks/src/main/arm/scripts/inline-scripts/enableHpa.sh
    az aks get-credentials \
./weblogic-azure-aks/src/main/arm/scripts/inline-scripts/enablePrometheusMetrics.sh
    az aks get-credentials \
./weblogic-azure-aks/src/main/arm/scripts/updateApplications.sh
    az aks get-credentials \
./weblogic-azure-aks/src/main/arm/scripts/validateApplications.sh
    az aks get-credentials --resource-group ${AKS_CLUSTER_RESOURCEGROUP_NAME} --name ${AKS_CLUSTER_NAME} --overwrite-existing
./weblogic-azure-aks/src/main/arm/scripts/setupWLSDomain.sh
  az aks get-credentials \
./weblogic-azure-aks/src/main/arm/scripts/queryStorageAccount.sh
    az aks get-credentials \
./weblogic-azure-aks/src/main/arm/scripts/setupDBConnections.sh
  az aks get-credentials --resource-group ${AKS_CLUSTER_RG_NAME} --name ${AKS_CLUSTER_NAME} --overwrite-existing
./weblogic-azure-aks/src/main/arm/scripts/setupNetworking.sh
```

File: weblogic-azure-aks/src/main/bicep/modules/_deployment-scripts/_ds_enable_hpa.bicep [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+resource deploymentScript 'Microsoft.Resources/deploymentScripts@${azure.apiVersionForDeploymentScript}' = {
+  name: const_deploymentName
+  location: location
+  kind: 'AzureCLI'
+  identity: identity
+  properties: {
+    azCliVersion: azCliVersion
+    scriptContent: format('{0}\r\n\r\n{1}\r\n\r\n{2}',base64ToString(base64_common), base64ToString(base64_utility), base64ToString(base64_enableHpa))
+    environmentVariables: [
```
Comment (edburns) [inline weblogic-azure-aks/src/main/bicep/modules/_deployment-scripts/_ds_enable_hpa.bicep:33]: Very clever.

File: weblogic-azure-aks/src/main/arm/scripts/inline-scripts/enablePrometheusMetrics.sh (hunk 3 above)
Comment (galiacheng) [reply weblogic-azure-aks/src/main/arm/scripts/inline-scripts/enablePrometheusMetrics.sh]: See https://github.com/oracle/weblogic-azure/pull/297/commits/d6cb492769ef50fffcb502cd3d6c11ff741fd7e8

Comment (edburns) [reply weblogic-azure-aks/src/main/arm/scripts/inline-scripts/enablePrometheusMetrics.sh]: Thanks. I'll mark this as resolved.

**Review by edburns — APPROVED** [review]
This all works. Please merge it so we can make it live!

**Review by rjeberhard — APPROVED** [review]

**Pull request merged**